client/ayon_deadline/addon.py

```diff
@@ -46,47 +52,233 @@ def get_plugin_paths(self):
+            webservice_url (str): Server url.
+            auth (Optional[Tuple[str, str]]): Tuple containing username,
+                password
+            log (Optional[Logger]): Logger to log errors to, if provided.
+
+        Returns:
+            List[str]: Limit Groups.
+
+        Raises:
+            RuntimeError: If deadline webservice_url is unreachable.
+
+        """
+        endpoint = "{}/api/limitgroups?NamesOnly=true".format(webservice_url)
+        return DeadlineAddon._get_deadline_info(
+            endpoint, auth, log, item_type="limitgroups")
+
+    @staticmethod
+    def get_deadline_workers(
+        webservice_url: str,
+        auth: Optional[Tuple[str, str]] = None,
+        log: Optional[Logger] = None
+    ) -> List[str]:
+        """Get Workers (eg.machine names) from Deadline API.
+
+        Args:
+            webservice_url (str): Server url.
+            auth (Optional[Tuple[str, str]]): Tuple containing username,
+                password
+            log (Optional[Logger]): Logger to log errors to, if provided.
+
+        Returns:
+            List[str]: Limit Groups.
+
+        Raises:
+            RuntimeError: If deadline webservice_url is unreachable.
+
+        """
+        endpoint = "{}/api/slaves?NamesOnly=true".format(webservice_url)
+        return DeadlineAddon._get_deadline_info(
+            endpoint, auth, log, item_type="workers")
+
+    @staticmethod
+    def _get_deadline_info(endpoint, auth=None, log=None, item_type=None):
         from .abstract_submit_deadline import requests_get
 
         if not log:
             log = Logger.get_logger(__name__)
 
-        argument = "{}/api/pools?NamesOnly=true".format(webservice)
         try:
             kwargs = {}
             if auth:
                 kwargs["auth"] = auth
-            response = requests_get(argument, **kwargs)
+            response = requests_get(endpoint, **kwargs)
         except requests.exceptions.ConnectionError as exc:
-            msg = 'Cannot connect to DL web service {}'.format(webservice)
+            msg = 'Cannot connect to DL web service {}'.format(endpoint)
             log.error(msg)
             six.reraise(
                 DeadlineWebserviceError,
                 DeadlineWebserviceError('{} - {}'.format(msg, exc)),
                 sys.exc_info()[2])
         if not response.ok:
-            log.warning("No pools retrieved")
+            log.warning(f"No {item_type} retrieved")
             return []
 
         return response.json()
+
+    def pools_by_server_name(self, server_name: str) -> List[str]:
+        """Returns dictionary of pools per DL server
+
+        Args:
+            server_name (str): Deadline Server name from Project Settings.
+
+        Returns:
+            Dict[str, List[str]]: {"default": ["pool1", "pool2"]}
+
+        """
+        pools = self._pools_by_server_name.get(server_name)
+        if pools is None:
+            dl_server_info = self.deadline_servers_info.get(server_name)
```

Comment:
BTW all the methods have this "safe" call of `get`, but it is not validated and expected the server name is there? We should either change it to `dl_server_info = self.deadline_servers_info[server_name]` or return empty list if `dl_server_info` is `None`.